litert/vendors/qualcomm/core/wrappers/tensor_wrapper.h

```diff
@@ -112,7 +112,9 @@ class TensorWrapper final {
 
   ~TensorWrapper();
 
-  bool operator==(const TensorWrapper& other) const { return this == &other; }
+  bool operator==(const TensorWrapper& other) const {
+    return CompareTensor(qnn_tensor_, other.qnn_tensor_);
```

Comment:
We can still check the address first.
```suggestion
    return this == &other || CompareTensor(qnn_tensor_, other.qnn_tensor_);
```